### DISCUSSION TITLE
Question about SQL Query to delete images in sqlite database.

### DISCUSSION THREAD

**Discussion by meteos77** [textbrowser/biblioteq#341] [Q&A] [answered]
After uploading the cover images, my database is 110MB in size.
I now want to remove the images to get a lighter database.

I've created 2 queries but I can't really see the result :

`UPDATE book SET front_cover= ""`
and
`DELETE front_cover FROM book`

+ vaccum for the 2 bases

with the UPDATE command, my database is 2.7MB in size
on the DELETE command, my database is 5.4MB in size

Why is my database after DELETE twice as large as with UPDATE?

**Comment by textbrowser** [comment]
The BYTEA type must not be populated with a string. Set the columns to NULL. If a column allows for a NULL, set a NULL if that's the intent. Don't set it to an empty string if you desire a NULL. An empty string is not a NULL string.

You cannot delete a column from a table so the query is nonsense. You can drop a column but that's destructive.

**Reply by meteos77** [reply, marked as answer]
UPDATE book SET front_cover = NULL

thank you for your advice !

**Comment by textbrowser** [comment]
An empty string will not be discoverable if a query intends to discover NULL values. You'll need to test the length of the string in addition to NULL.

I recommend purchasing a book on SQL and reading it thoroughly instead of using discussions as personal forever-present assistants. :P

**Reply by meteos77** [reply]
Thank you for taking the time to reply.
I'm starting SQL, I'm going to start a book even if it's more difficult than your explanations.

**Comment by textbrowser** [comment]
Languages cannot be simple if they want to exchange complex ideas.

**Comment by textbrowser** [comment]
Like I'm sure that it's impossible to translate a Vietnamese poem into French while retaining the pitch of the poem. SQL isn't easy but that's because data possibilities are infinite. More practice, better results.

**Reply by meteos77** [reply]
With QH, we discover a bit of SQL (the queries hidden by BQ, which is very instructive)
